connectors/cache/fallback.go

```diff
@@ -0,0 +1,272 @@
+package cache
+
+import (
+	"context"
+	"errors"
+	"sort"
+
+	"github.com/uber-go/dosa"
+	"github.com/uber-go/dosa/connectors/base"
+)
+
+const (
+	key   = "key"
+	value = "value"
+)
+
+type rangeResults struct {
+	Rows      []map[string]dosa.FieldValue
+	TokenNext string
+}
+
+type rangeQuery struct {
+	Conditions map[string][]*dosa.Condition `json:",omitempty"`
+	Token      string
+	Limit      int
+}
+
+// NewConnector creates a fallback cache connector
+func NewConnector(origin dosa.Connector, fallback dosa.Connector, encoder Encoder, entities ...dosa.DomainObject) *Connector {
+	bc := base.Connector{Next: origin}
+	set := createCachedEntitiesSet(entities)
+	return &Connector{
+		Connector:         bc,
+		origin:            origin,
+		fallback:          fallback,
+		encoder:           encoder,
+		cacheableEntities: set,
+	}
+}
+
+// Connector is a fallback cache connector
+type Connector struct {
+	base.Connector
+	origin            dosa.Connector
+	fallback          dosa.Connector
+	encoder           Encoder
+	cacheableEntities map[string]bool
+	// Used primarily for testing so that nothing is called in a goroutine
+	synchronous bool
+}
+
+// SetCachedEntities sets the entities that will write and read from the fallback
+func (c *Connector) SetCachedEntities(entities ...dosa.DomainObject) {
+	set := createCachedEntitiesSet(entities)
+	c.cacheableEntities = set
```

Comment:
is it ok that this isn't thread-safe? this should only be called once on initialization or no?